oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/RelatedSelectField/ExternalApiModal.jsx

```diff
@@ -0,0 +1,230 @@
+  fieldPath,
+  serializeSuggestions,
+}) => {
+  const [externalApiRecord, setExternalApiRecord] = useState({});
+  const { setFieldValue } = useFormikContext();
+  const searchApi = new InvenioSearchApi(searchConfig.searchApi);
+  const handleExternalRecordChange = (record) => {
+    setExternalApiRecord(record);
+  };
+  return (
+    <Modal open={open} onClose={onClose}>
+      <Modal.Header as="h6" className="pt-10 pb-10">
+        <Grid>
+          <Grid.Column floated="left">
+            <Header as="h2">Search results from external API</Header>
+          </Grid.Column>
+        </Grid>
+      </Modal.Header>
+      <Modal.Content>
+        <OverridableContext.Provider value={overriddenComponents}>
+          <ReactSearchKit
+            searchApi={searchApi}
+            initialQueryState={searchConfig.initialQueryState}
+            urlHandlerApi={{ enabled: false }}
+          >
+            <Grid celled="internally">
+              <Grid.Row>
+                <Grid.Column width={8} floated="left" verticalAlign="middle">
+                  <SearchBar
+                    placeholder={i18next.t("Search")}
+                    autofocus
+                    actionProps={{
+                      icon: "search",
+                      content: null,
+                      className: "search",
+                    }}
+                  />
+                </Grid.Column>
+              </Grid.Row>
+              <Grid.Row verticalAlign="middle">
+                <Grid.Column>
+                  <ResultsLoader>
+                    <EmptyResults />
+                    <Error />
+                    <ExternalApiResultsList
+                      handleAddingExternalApiSuggestion={
+                        handleAddingExternalApiSuggestion
+                      }
+                      fieldPath={fieldPath}
+                      handleExternalRecordChange={handleExternalRecordChange}
+                      externalApiRecord={externalApiRecord}
+                      serializeSuggestions={serializeSuggestions}
+                    />
+                  </ResultsLoader>
+                </Grid.Column>
+              </Grid.Row>
+              <Grid.Row verticalAlign="middle">
+                <Grid.Column>
+                  <Pagination options={{ size: "tiny" }} />
+                </Grid.Column>
+
+                <Grid.Column floated="right" width={3}>
+                  <ResultsPerPage
+                    values={searchConfig.paginationOptions.resultsPerPage}
+                    label={resultsPerPageLabel}
+                  />
+                </Grid.Column>
+              </Grid.Row>
+            </Grid>
+          </ReactSearchKit>
+        </OverridableContext.Provider>
+      </Modal.Content>
+      <Modal.Actions>
+        <Button
+          name="cancel"
+          onClick={() => {
+            onClose();
+          }}
+          icon="remove"
+          labelPosition="left"
+          content={i18next.t("Cancel")}
```

Comment:
Better would be to use "Close" as there is not really any process to cancel. Can be also moved top right and implemented as "X" close button.